### ISSUE TITLE
renewcommand fails for builtin command

### ISSUE THREAD

**Issue by PatrickMassot** [plastex/plastex#90]
Currently ``\renewcommand`` works only for user defined commands but not builtin commands. Consider the following input:
```latex
\documentclass{article}
\renewcommand{\textbf}[1]{YOP}
\begin{document}
\textbf{Test}.
\end{document}
```
When complied by pdflatex, you see YOP. But plasTeX (master branch, default config) displays **Test**.
This is due to [these lines](https://github.com/tiarno/plastex/blob/f07aab870059c1b548e54c0a7b17482884cc9060/plasTeX/Context.py#L945-L946).

**Comment by kesmit13** [comment]
Yes.  This is by design.  I've always made the assumption that if there is a Python-implemented class to handle a macro, that it was hand-tuned to work better than a LaTeX implementation of that same macro.  So I don't allow a LaTeX macro to redefine a Python class-based macro.

**Comment by PatrickMassot** [comment]
I see. I won't discuss that design choice, but I think it should be documented somewhere (it took me a *long* time to understand what was going on). You could also mention that using ``\def`` is not forbidden in this context.

**Comment by kesmit13** [comment]
I'll admit plasTeX gets into some weird areas when trying to merge the worlds of (La)TeX macros vs. Python class, linear documents vs. DOM objects, etc.  Things don't always fit together perfectly.  I don't recall why ``\def`` was handled differently though.

**Comment by dalcde** [comment]
I also find this behavior counter-intuitive. When I use `\renewcommand`, I almost always mean "I want to define a command with this name. There is already a command that uses this name but I never use it, so let me overwrite it", as opposed to "I have a better definition of this command that performs the same function".

I propose the following behaviour:
1. `\newcommand` *never* overwrites any existing definition. If there is an existing python definition, we print a warning and keep the python definition. If there is an existing latex definition, it throws an error. This avoids the problem in #22 .
2. `\renewcommand` by default always overwrites any existing definition.
3. A python implementation that doesn't want to be overwritten can set a `forceKeep` parameter to avoid being overwritten by `\renewcommand`.

The main problem I can foresee is that some packages may want to redefine a command to be compatible with their package, e.g. `amsthm` redefines `newtheorem`. However, in my experience, the packages are usually sufficiently complex that we will usually have a python redefinition of the package and the latex will not be parsed.

**Comment by PatrickMassot** [comment]
@kesmit13 Do you mind if we follow @dalcde 's suggestion?

**Comment by PatrickMassot** [comment]
@dalcde I think you can move forward if you want to implement this.

**Comment by PatrickMassot** [comment]
@dalcde I know I write this every six month but I would really like to release plasTeX 3.0 soon. So now is a good time to introduce backward incompatible changes. Do you think you could implement the changes discussed in this issue?

**Comment by PatrickMassot** [comment]
I ended up implementing a simpler behavior, without trying to figure out whether a definition comes from LaTeX or python.

**Comment by christianp** [comment]
Can I reopen this? A Chirun user asked why they couldn't redefine `\varpi` and after tracing backwards through the code and commit log, I found this.

It looks like @PatrickMassot 's commit 0e147436a4973a1fa484783b9371894ce4720765 was reverted in eed8d46758c6b7d3b26385d4844bdbf8f42bf9cb. I can't see why - there was no explanation in the commit message.

I guess that the chaos that ensued from LaTeX packages redefining stuff defined by plasTeX was too much trouble.

For now, I've added in a hack to solve our immediate problem, adding a `redefinable` attribute to the `Macro` class, which is true for all subclasses of `MathSymbol`. That's in my branch [`redefine-math-symbols`](https://github.com/plastex/plastex/compare/master...christianp:plastex:redefine-math-symbols). I'm not at all sure that this is the right way to solve this problem in general, so I haven't made a PR.

**Comment by PatrickMassot** [comment]
We can definitely reopen this (and I’ll do it right now). I still think this is a very confusing behavior of plasTeX. Unfortunately I don’t remember why I reverted this (there is a reason why we should write better commit messages…). But probably it was creating a mess somewhere else. I’ll put that on the pile of things to think about, maybe during Christmas vacation.

**Comment by christianp** [comment]
Just adding a note that I've seen a document where the author did `\renewcommand{\thechapter}{\Roman{chapter}}`, so plasTeX's refs for TOC entries ended up not matching pdflatex's because it was still producing decimal chapter numbers.
